File: labdapattogas/labdapattogas.php
<?php

  function gotoxy($x, $y) {
      for($i=0; $i < $y; $i++) {
        echo"\n";
      }

      for($i=0; $i < $x; $i++) {
        echo" ";
      }

      echo"o\n";
  }

  $x = 10;
  $y = 20;

  $xX = 1;
  $yY = -1;
  
  $yArray = array();
  $xArray = array();

  for($i = 0; $i < 23; $i++) {
    array_push($yArray, 1);
  }

  $replaceyArray = array(1 => -1, 22 => -1);
  $yArray = array_replace($yArray, $replaceyArray);


  for($i = 0; $i < 80; $i++) {
    array_push($xArray, 1);
  }

  $xArrayDo = array(1 => -1, 79 => -1);
  $xArray = array_replace($xArray, $xArrayDo);


  for(;;) {

    for($i=0; $i < 37; $i++) {
      echo"_";
    }

    echo "x=",$x,"y=",$y;

    for($i=0; $i < 37; $i++) {
      echo"_";
    }

    gotoxy($x, $y);

    $x += $xX;
    $y += $yY;

    $xX *= $xArray[$x];
    $yY *= $yArray[$y];

    sleep(1);
    system('clear');

  }

?>

File: labdapattogas/labdapattogas.py
import os
from time import sleep

def sClear():
    os.system('sClear' if os.name=='nt' else 'clear')

def gotoxy(x,y):
    for i in range(y):
        print()
    for i in range(x):
        print(" ", end="")
    print("o")

x = 10
y = 20

xx = 1
yy = -1

yArray = []
xArray = []

for i in range(23):
    yArray.append(1)

yArray[1] = -1
yArray[22] = -1


for i in range(80):
    xArray.append(1)


xArray[1] = -1
xArray[79] = -1


while (1):
    for i in range(37):
        print("_", end="")

    print("x=", x, end="")
    print("y=", y, end="")

    for i in range(37):
        print("_", end="")

    gotoxy(x,y)

    x+=xx
    y+=yy

    xx*=xArray[x]
    yy*=yArray[y]

    sleep(0.5)
    sClear()